### PR TITLE
Rename BenchOpt to Benchopt

### LINKED ISSUE #249: cosmetic: rename to benchopt in lowercase?
We have BenchOpt, Benchopt and benchopt appearing in the doc, the GH organization is named benchopt while the main repo is benchOpt.

My 2 cents is that all lower case is easier. It is far from being critical.

### PR DESCRIPTION
**Pull request by mathurinm**
closes #249 

I have used `Benchopt` when it's at the start of a sentence, and `benchopt` otherwise

### CODE REVIEW

File: doc/how.rst [hunk 1]
```diff
@@ -147,7 +147,7 @@ This ``stop_strategy`` can be:
       compute and store the objective and return ``False`` once the computations
       should stop.
 
-BenchOpt supports different types of solvers:
+Benchopt supports different types of solvers:
```
Comment (agramfort) [inline doc/how.rst:150]:
```suggestion
benchopt supports different types of solvers:
```
?

Comment (mathurinm) [reply doc/how.rst:150]: beginning of sentence: capitalize
else: all lower case

WDYT?

Comment (tomMoral) [reply doc/how.rst:150]: I am +1 for @mathurinm  solution.

**Review by tomMoral — APPROVED** [review]
LGTM!

**Pull request merged**